File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.py
def solution(my_string, queries):
    ls = list(my_string)
    for query in queries:
        a, b = query[0], query[1]
        temp = ls[a:b+1]
        temp = temp[::-1]
        ls = ls[:a] + temp + ls[b+1:]
    out_str = "".join(ls)
    return out_str

File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(string my_string, vector<vector<int>> queries) {
    for (const auto& query : queries) {
        int s = query[0];
        int e = query[1];
        
        while (s < e) {
            swap(my_string[s], my_string[e]);
            s++;
            e--;
        }
    }
    
    return my_string;
}